src/Makefile.inc

```diff
@@ -0,0 +1,340 @@
+
+ifdef MODULAR
+PROTO_OBJS  := $(addprefix $(ODIR)/, $(notdir $(PROTO_SRC_C:.c=.o)))
+PROTO_LIBS  := $(addprefix $(ODIR)/, $(notdir $(PROTO_SRC_C:.c=.bin)))
+PROTO_EXTRA_OBJS := $(addprefix $(ODIR)/, $(notdir $(PROTO_EXTRA_C:.c=.o_)))
+SYMBOL_FILE := $(ODIR)/symbolfile
+endif
+
+HGVERSION ?= $(shell $(SDIR)/../utils/get_version.pl ${TARGET})
+HGVERSION := $(HGVERSION)
+
+# Rebuild on hg version change.
+# Based on http://mercurial.selenic.com/wiki/VersioningWithMake
+# We need to always run this code, as opposed to running it from a prerequisite
+# The HGTEST variable is not used
+HGTEST := $(shell mkdir -p $(ODIR); \
+	[ -f $(ODIR)/hgstamp ] || touch $(ODIR)/hgstamp; \
+	echo $(HGVERSION) | cmp -s $(ODIR)/hgstamp - \
+	|| echo $(HGVERSION) > $(ODIR)/hgstamp)
+############################################
+#END SECTION                               #
+############################################
+
+##################################################
+#This section contains switches used for building#
+##################################################
+ifeq "$(TYPE)" "dev"
+  CFLAGS   := $(CFLAGS) -DBUILDTYPE_DEV
+endif
+ifdef MODULAR
+MODULE_CFLAGS   :=  -DMODULAR=$(MODULAR)
+CFLAGS   +=  -DENABLE_MODULAR=$(MODULAR)
+endif
+
+CFLAGS   := $(CFLAGS) -g -Wall -Wextra -Werror=undef -I$(SDIR) -I$(SDIR)/firmware -D$(PROGMODE) -std=gnu99 \
+            -I$(SDIR)/target/tx/$(FAMILY)/$(TARGET) -I$(SDIR)/gui/$(SCREENSIZE) \
+            -I$(SDIR)/pages/$(SCREENSIZE) \
+            -DHGVERSION="\"${HGVERSION}\""
+CXXFLAGS := $(CXXFLAGS) $(patsubst -std=gnu99,,$(CFLAGS))
+EXEEXT   ?= elf
+##################################################
+#END SECTION                                     #
+##################################################
+
+############################################
+#this section defines final build files    #
+############################################
+FILESYSTEM = $(subst emu_,,$(TARGET))
+MODELDIR = filesystem/$(FILESYSTEM)/models
+
+LAST_MODEL := $(MODELDIR)/model$(NUM_MODELS).ini
+
+ALL += $(TARGET).fs_wrapper
+
+############################################
+#END SECTION                               #
+############################################
+
+############################################
+#this section includes build targets       #
+############################################
+all : $(ALL)
+
+####################################
+# recompile if the Makefile changes#
+####################################
+#$(OBJS) $(PROTO_OBJS) ($PROTO_EXTRA_OBJS): Makefile target/tx/$(FAMILY)/$(TARGET)/Makefile.inc $(EXTRA_MAKEFILES)
+
+##################################################################################
+# The following enables quiet output unless you use VERBOSE=1                    #
+# Note that this must be after the 1st rule so that it doesn't execute by default#
+##################################################################################
+$(VERBOSE).SILENT:
+
```

Comment:
This is one of the thinks I hate about Makefiles.  why does the order of things like this matter?  it is completely non-intuitive